clang/test/Analysis/Checkers/WebKit/uncounted-lambda-captures.cpp

```diff
@@ -299,6 +300,29 @@ struct RefCountableWithLambdaCapturingThis {
       return obj->next();
     });
   }
+
+  void callAsyncNoescape([[clang::noescape]] WTF::Function<bool(RefCountable&)>&&);
+  void method_temp_lambda(RefCountable* obj) {
+    callAsyncNoescape([this, otherObj = RefPtr { obj }](auto& obj) {
+      return otherObj == &obj;
+    });
+  }
+
+  void method_nested_lambda() {
+    callAsync([this, protectedThis = Ref { *this }] {
+      callAsync([this, protectedThis = static_cast<const Ref<RefCountableWithLambdaCapturingThis>&&>(protectedThis)] {
+        nonTrivial();
+      });
+    });
+  }
+
+  void method_nested_lambda2() {
+    callAsync([this, protectedThis = RefPtr { this }] {
+      callAsync([this, protectedThis = static_cast<const Ref<RefCountableWithLambdaCapturingThis>&&>(*protectedThis)] {
+        nonTrivial();
+      });
+    });
+  }
```

Comment:
Do we need a test for a lambda capture with a `this` and an unprotected `this`?

Reply:
Oh, we have such a test case. See method_captures_this_unsafe around line 217 :)

Reply:
I guess we can add a test case for nested lambdas though.

Reply:
Added a test case.